src/app/public/src/modules/edit-button/edit-button.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { StacheJsonDataService, StacheRouteService } from '../shared';
+import { StacheNavService } from '../nav';
+
+@Component({
+  selector: 'stache-edit-button',
+  templateUrl: './edit-button.component.html',
+  styleUrls: ['./edit-button.component.scss']
+})
+export class StacheEditButtonComponent implements OnInit {
+  @Input()
+  public editButtonText = 'Edit';
+
+  private url: string;
+
+  public constructor (
+    private dataService: StacheJsonDataService,
+    private navService: StacheNavService,
+    private routeService: StacheRouteService
+  ) { }
+
+  public ngOnInit() {
+    this.url = this.getUrl();
+  }
+
+  public onEditButtonClick() {
+    this.navService.navigate({ path: this.url });
+  }
+
+  private getUrl(): string {
+    const repoData = this.dataService.getByName('repo');
+    let frag = encodeURIComponent(this.routeService.getActiveUrl() + '/');
+    let result = repoData.url + frag + 'index.html';
+
+    return repoData.vstsBranchSelector === undefined ? result : result + repoData.vstsBranchSelector;
```

Comment:
Again, non-blocking style comment: parenthesis are a good indicator that an expression should be evaluated as boolean: `(repoData.vstsBranchSelector === undefined)`.